crates/admin/src/cluster_controller/service.rs

```diff
@@ -760,17 +674,25 @@ mod tests {
             let lsn = appender.append(format!("record{i}")).await?;
             assert_eq!(Lsn::from(i), lsn);
         }
-
-        // report persisted lsn back to cluster controller for a subset of the nodes
-        persisted_lsn.store(5, Ordering::Relaxed);
-
+        // archived_lsn.store(5, Ordering::Relaxed);
         tokio::time::sleep(interval_duration * 10).await;
-        // no trimming should have happened because one node did not report the persisted lsn
+
+        // no trimming should have happened because no nodes report archived_lsn
         assert_eq!(Lsn::INVALID, bifrost.get_trim_point(LOG_ID).await?);
 
         Ok(())
     }
 
+    #[test(restate_core::test(start_paused = true))]
+    async fn do_not_trim_if_dead_nodes_present() -> anyhow::Result<()> {
```

Comment:
What if a node has permanently been decommissioned?

Reply:
This must be dealt with somehow; I think this is telling us that we're missing an operational tool to explicitly remove permanently decommissioned nodes. We might also want to add a tolerance level to the auto-trim component for dead/unknown/lagging nodes.